### PR TITLE
Add oVirt ovirt_nics and ovirt_nics_facts modules

### PR DESCRIPTION
**Pull request by machacekondra**
##### ISSUE TYPE

- New Module Pull Request<
##### COMPONENT NAME

- ovirt_nics
- ovirt_nics_facts
##### SUMMARY

This PR adds two new modules for managing VM's network interfaces.

### CODE REVIEW

**Comment by machacekondra** [comment]
@mwperina @ryansb

**Comment by gregdek** [comment]
Thanks again to @machacekondra. This module is going into community review.

We notice that you are one of the maintainers (@machacekondra) of this module, so when you think it's ready, please comment "shipit" and we will consider it for merging.

[This message brought to you by your friendly Ansibull-bot.]

File: cloud/ovirt/ovirt_nics_facts.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        description:
+            - "Name of the NIC, can be used as glob expression."
+extends_documentation_fragment: ovirt
+'''
+
+EXAMPLES = '''
+# Examples don't contain auth parameter for simplicity,
+# look at ovirt_auth module to see how to reuse authentication:
+
+# Gather facts about all NICs named C<eth*>:
```
Comment (mwperina) [inline cloud/ovirt/ovirt_nics_facts.py]: I'd rephrase to: 'Gather facts about all NICs which names start with C<eth>:'

**Review by mwperina — CHANGES_REQUESTED** [review]
I'd also replace 'Virtual Machine' with 'VM' to be consistent but not a must

File: cloud/ovirt/ovirt_nics_facts.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        description:
+            - "Name of the NIC, can be used as glob expression."
+extends_documentation_fragment: ovirt
+'''
+
+EXAMPLES = '''
+# Examples don't contain auth parameter for simplicity,
+# look at ovirt_auth module to see how to reuse authentication:
+
+# Gather facts about all NICs which names start with C(eth*):
```
Comment (mwperina) [inline cloud/ovirt/ovirt_nics_facts.py]: 'C(eth*)' -> 'C(eth)'

**Review by mwperina — CHANGES_REQUESTED** [review]

File: cloud/ovirt/ovirt_nics_facts.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        description:
+            - "Name of the NIC, can be used as glob expression."
+extends_documentation_fragment: ovirt
+'''
+
+EXAMPLES = '''
+# Examples don't contain auth parameter for simplicity,
+# look at ovirt_auth module to see how to reuse authentication:
+
+# Gather facts about all NICs which names start with C(eth):
```
Comment (mwperina) [inline cloud/ovirt/ovirt_nics_facts.py]: I forgot that vm name is also specified, so the comment should be:

Gather facts about all NICs which names start with C(eth) for VM named C(centos7)

File: cloud/ovirt/ovirt_nics_facts.py [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+- ovirt_nics_facts:
+    vm: rhel7
```
Comment (mwperina) [inline cloud/ovirt/ovirt_nics_facts.py]: 'rhel7' -> 'centos7'

**Review by mwperina — CHANGES_REQUESTED** [review]

File: cloud/ovirt/ovirt_nics_facts.py (hunk 4 above)
Comment (machacekondra) [reply cloud/ovirt/ovirt_nics_facts.py]: done

File: cloud/ovirt/ovirt_nics_facts.py (hunk 3 above)
Comment (machacekondra) [reply cloud/ovirt/ovirt_nics_facts.py]: done

**Review by mwperina — APPROVED** [review]
Looks good to me, thanks! :+1:

**Comment by machacekondra** [comment]
shipit

**Comment by gregdek** [comment]
Thanks @machacekondra. Since you are one of the maintainers (@machacekondra) of this module and you commented with "shipit", we are marking this PR for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
Thanks again to @machacekondra. This module is going into community review.

We notice that you are one of the maintainers (@machacekondra) of this module, so when you think it's ready, please comment "shipit" and we will consider it for merging.

[This message brought to you by your friendly Ansibull-bot.]

File: cloud/ovirt/ovirt_nics_facts.py [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+        module.exit_json(
+            changed=False,
+            ansible_facts=dict(
+                ovirt_nics=[
+                    get_dict_of_struct(c) for c in nics
+                ],
+            ),
+        )
+    except Exception as e:
+        module.fail_json(msg=str(e))
```
Comment (ryansb) [inline cloud/ovirt/ovirt_nics_facts.py]: Same here, please surface the full exception.

File: cloud/ovirt/ovirt_nics.py [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+            vmnics_module.create()
+            ret = vmnics_module.action(
+                action='deactivate',
+                action_condition=lambda nic: nic.plugged,
+                wait_condition=lambda nic: not nic.plugged,
+            )
+
+        module.exit_json(**ret)
+    except Exception as e:
+        module.fail_json(msg=str(e))
```
Comment (ryansb) [inline cloud/ovirt/ovirt_nics.py]: Same here, please surface the full exception.

**Comment by machacekondra** [comment]
shipit

**Comment by gregdek** [comment]
Thanks @machacekondra for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by machacekondra** [comment]
shipit

**Comment by gregdek** [comment]
Thanks @machacekondra. Since you are one of the maintainers (@machacekondra) of this module and you commented with "shipit", we are marking this PR for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Review by mwperina — APPROVED** [review]
:+1:
